### PR TITLE
Make some changes to OC Calculator to add functionality to be used by God Forge

### PR DESCRIPTION
**Pull request by BlueWeabo**
Allows one to directly put in a double to have an OC like 4.1 of energy increase or 2.1 duration decrease.

This allows a for a bit more customization and upgrades for the future of multiblock

### CODE REVIEW

**Review by GDCloudstrike — APPROVED** [review]
Thanks

**Review by miozune — CHANGES_REQUESTED** [review]
I think we should deprecate old integer methods to reduce confusion. Also I'd guess your editor has _strong_ opinion on max amount of characters per line...

File: src/main/java/gregtech/api/util/GT_OverclockCalculator.java [hunk 1]
```diff
@@ -277,37 +275,71 @@ public GT_OverclockCalculator setHeatDiscountMultiplier(float heatDiscountExpone
     }
 
     /**
-     * Sets the Overclock that should be calculated when one. This uses BitShifting! Default is 2, which is a 4x
-     * decrease
+     * Sets the Overclock that should be calculated when a heatOC is applied. Calls {@link #setHeatPerfectOC(double)}
+     * where the given value is 2^heatPerfectOC
      */
+    @Deprecated
```
Comment (miozune) [inline src/main/java/gregtech/api/util/GT_OverclockCalculator.java:281]: In general adding `@deprecated` javadoc with describing what alternative should be used, would be the right way to apply `@Deprecated` annotation

**Review by miozune — APPROVED** [review]
Thanks for addressing my nitpicks

**Comment by BlueWeabo** [comment]
Good to have then especially when i had written everything at midnight thinking it was a 10 min job turning into 1 hour :clueless:

**Pull request merged**